### DISCUSSION TITLE
Is there a grid like layout in giu?

### DISCUSSION THREAD

**Discussion by Seikaijyu** [AllenDang/giu#530] [Q&A] [answered]
If not, what should I use instead?
Will grid layout be considered in subsequent versions?

**Comment by gucio321** [comment, marked as answer]
Hi there!
what do you mean by "Grid Layout"?

**Reply by Seikaijyu** [reply]
Just like the 
```xml
<Grid></Grid> 
```
in WPF

**Reply by Seikaijyu** [reply]
An adaptive layout, which can modify the width and height coefficient of a grid and automatically adapt to the window layout

**Reply by Seikaijyu** [reply]
If you have used WPF, you will know how easy this layout is to use. It is omnipotent

**Reply by gucio321** [reply]
well, so it's just a table, right?
![image](https://user-images.githubusercontent.com/73652197/179411243-7b763fd4-2c6f-4468-a798-25754b1cfb5c.png)

**Reply by Seikaijyu** [reply]
There are still many differences with tables

**Reply by Seikaijyu** [reply]
Can I achieve a similar effect in giu?

**Reply by Seikaijyu** [reply]
For details, see the demonstration I give below, especially the cross column lattice and customized relative size

**Reply by Seikaijyu** [reply]
Or try to introduce the following code into your Xaml
```xml
 <Grid>
        <Grid.RowDefinitions>
            <RowDefinition Height="0.4*"/>
            <RowDefinition Height="0.2*"/>
        </Grid.RowDefinitions>
        <Grid.ColumnDefinitions>
            <ColumnDefinition Width="0.5*"/>
            <ColumnDefinition Width="0.2*"/>
        </Grid.ColumnDefinitions>
        <Border Grid.Row="0" Grid.Column="0" Background="Red"/>
        <Border Grid.Row="0" Grid.RowSpan="2" Grid.Column="1" Background="Blue"/>
        <Border Grid.Row="1" Grid.Column="0" Background="Yellow"/>
    </Grid>
```

**Reply by gucio321** [reply]
hmm, could you run examples/widgets and go to `Group` tab - is it what you need?
I'll give some more specified demo in a while ;-)

**Reply by gucio321** [reply]
ok it isn't that - it doesn't fit window
so I'd suggest tables but need to find the best variant to fit height

**Reply by gucio321** [reply]
for now tables works like that:
```golang
package main

import "github.com/AllenDang/giu"

func main() {
	wnd := giu.NewMasterWindow("discussion 530", 640, 480, 0)
	wnd.Run(loop)
}

func loop() {
	giu.SingleWindow().Layout(
		giu.Row(
			giu.Table().Flags(0).Rows(
				giu.TableRow(
					giu.Label("Row 1, column 1"),
					giu.Label("Row 1, column 2"),
				),
				giu.TableRow(
					giu.Label("Row 2, column 1"),
					giu.Label("Row 2, column 2"),
				),
			),
		),
	)
}
```
however I'm sure that row's height could be set as well

**Reply by gucio321** [reply]
mehh @AllenDang Flags.go doesn't have comments over each flag constant that makes understanding them a bit hard...

**Reply by Seikaijyu** [reply]
This looks good, but it doesn't seem to span multiple rows or columns
And no adaptation
Is this the best alternative at present

**Reply by gucio321** [reply]
Well for autodialing you can use some of table flags(i don't remember which one exactly).

**Reply by Seikaijyu** [reply]
Thank you for your reminder. I may have to spend some time looking for it now

**Reply by gucio321** [reply]
No problem! In ca case of any questions feel free to ping me here

**Reply by Seikaijyu** [reply]
Well, thank you very much.

**Comment by gucio321** [comment]
~~q~~ sorry - misclick

**Reply by Seikaijyu** [reply]
![image](https://user-images.githubusercontent.com/68497295/179411529-3fddaff9-07ef-4dac-9fad-de74c660c19b.png)
In this picture, you can see the power of grid layout

**Reply by Seikaijyu** [reply]
Not only is it easy to understand, but it also scales with the window
